examples/prometheus-kubernetes/main.go

```diff
@@ -0,0 +1,1286 @@
+							},
+							Regex: prometheus.MustNewRegexp("prometheus"),
+						},
+						{
+							Action: prometheus.RelabelKeep,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_pod_label_component",
+							},
+							Regex: prometheus.MustNewRegexp("alertmanager"),
+						},
+						{
+							Action: prometheus.RelabelDrop,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_pod_container_port_number",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// TODO: Is there a cleaner way of doing this?
+	serverConfigBytes := new(bytes.Buffer)
+	if _, err := serverConfigBytes.ReadFrom(encoding.YAML(serverConfig)); err != nil {
+		panic(err)
+	}
+
+	serverConfigMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "release-name-prometheus-server",
+			Labels: map[string]string{
+				"app": "prometheus",
+				"component": "server",
+			},
+		},
+		Data: map[string]string{
+			"prometheus.yml": serverConfigBytes.String(),
+		},
+	}
+
+	generator.Add("server-configmap.yaml", encoding.YAML(serverConfigMap))
+
+	serverClusterRole := rbacv1beta1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "release-name-prometheus-kube-state-metrics",
+			Labels: map[string]string{
+				"app": "prometheus",
+				"component": "server",
+			},
+		},
+		Rules: []rbacv1beta1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{
+					"nodes",
+					"nodes/proxy",
+					"services",
+					"endpoints",
+					"pods",
+					"ingresses",
+				},
+				Verbs: []string{
+					"list",
+					"watch",
+					"get",
+				},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{
+					"configmaps",
+				},
+				Verbs: []string{
+					"get",
+				},
+			},
+			{
+				APIGroups: []string{"extensions"},
+				Resources: []string{
+					"ingresses/status",
+					"ingresses",
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"watch",
+				},
+			},
+			{
+				NonResourceURLs: []string{"/metrics"},
+				Verbs: []string{"get"},
+			},
+		},
+	}
+
+	generator.Add("server-clusterole.yaml", encoding.YAML(serverClusterRole))
+
+	serverClusterRoleBinding := rbacv1beta1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "release-name-prometheus-kube-state-metrics",
+			Labels: map[string]string{
+				"app": "prometheus",
+				"component": "server",
+			},
+		},
+		RoleRef: rbacv1beta1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind: "ClusterRole",
+			Name: "release-name-prometheus-server",
+		},
+		Subjects: []rbacv1beta1.Subject{
+			{
+				Kind: rbacv1beta1.ServiceAccountKind,
+				Name: "release-name-prometheus-server",
+				Namespace: namespace,
+			},
+		},
+	}
+
+	generator.Add("server-clusterrolebinding.yaml", encoding.YAML(serverClusterRoleBinding))
+
+	serverDeployment := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "release-name-prometheus-server",
+			Labels: map[string]string{
+				"app": "prometheus",
```

Comment:
those labels has to be constant I think - that's the whole point of Go templating to not make a typo (: